src/utils/math.hpp

```diff
@@ -250,31 +144,19 @@ constexpr unsigned int count_leading_zeros_impl(N n, std::size_t w) {
  */
 template<typename N>
 constexpr unsigned int count_leading_zeros(N n) {
-#if defined(__GNUC__) || defined(__clang__)
-	// GCC’s `__builtin_clz` returns an undefined value when called with 0
-	// as argument.
-	// [<http://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html>]
-	if (n == 0) {
-		// Return the quantity of zero bits in `n` rather than
-		// returning that undefined value.
-		return static_cast<unsigned int>(bit_width(n));
+	static_assert(std::is_integral<N>::value);
+	const auto x = static_cast<std::make_unsigned_t<N>>(n);
+	constexpr auto mask = std::remove_cv_t<std::make_unsigned_t<N>>{1};
```

Comment:
Can simplify these two `make_unsigned_t`s with `decltype(x)`

Reply:
Do you mean keeping the first one as-is and simplifying the second one?

Reply:
All after the `x` declaration.

Reply:
So, the changes that are already in 8bc4e13? The only bit left is the other comment about whether to use `auto` or not?